metrics/src/db.rs

```diff
@@ -0,0 +1,67 @@
+use super::REGISTRY;
+use once_cell::sync::Lazy;
+use prometheus::{HistogramOpts, HistogramVec, Opts};
+
+const RAW_DATA_SIZE_BUCKETS: &[f64; 8] = &[
+    5e+2_f64, 1e+3_f64, 2e+3_f64, 5e+3_f64, 5e+4_f64, 5e+5_f64, 5e+6_f64, 5e+7_f64,
+];
+
+static FETCHED_RAW_DATA_SIZE: Lazy<HistogramVec> = Lazy::new(|| {
+    let counter = HistogramVec::new(
+        HistogramOpts {
+            common_opts: Opts::new(
+                "db_raw_data_size",
+                "Size (in bytes) of raw data fetched from the database.",
+            ),
+            buckets: Vec::from(RAW_DATA_SIZE_BUCKETS as &'static [f64]),
+        },
+        &["message_type"],
+    )
+    .expect("db_raw_data_size metric can't be created");
+    REGISTRY
+        .register(Box::new(counter.clone()))
```

Comment:
Is there a graceful shutdown use-case for the sidecar? If so, we might consider unregistering the metrics before the process is terminated. Not a big issue, though.

Reply:
All good, thanks :+1: